### ISSUE TITLE
Why is the splitting of train dataset hardcoded?

### ISSUE THREAD

**Issue by GowthamGottimukkala** [tianyu0207/RTFM#10]
While dividing the train set into two classes, why is it hardcoded that first 63 values are one class and the remaining are one? Aren't we supposed to split it using the labels file provided through `option.py`
https://github.com/tianyu0207/RTFM/blob/950243a2f7521d88033f6c164d4ba6307ecfab62/main.py#L19-L24
https://github.com/tianyu0207/RTFM/blob/950243a2f7521d88033f6c164d4ba6307ecfab62/dataset.py#L23-L34

**Comment by tianyu0207** [comment]
> While dividing the train set into two classes, why is it hardcoded that first 63 values are one class and the remaining are one? Aren't we supposed to split it using the labels file provided through `option.py`
> https://github.com/tianyu0207/RTFM/blob/950243a2f7521d88033f6c164d4ba6307ecfab62/main.py#L19-L24
> 
> 
> https://github.com/tianyu0207/RTFM/blob/950243a2f7521d88033f6c164d4ba6307ecfab62/dataset.py#L23-L34

Yes. I got lazy and hard-coded the weak label for the training set. For the array [:63], all the videos are abnormal and the remaining are normal. This is just a function to provide abnormal/normal list for the dataloader so that I can evenly sample 32 videos for each batch from both classes.

**Comment by GowthamGottimukkala** [comment]
Ok thanks, I'll modify it so that the function divides it basing on the lablel.npy provided. By the way, does that mean the shanghai tech i3d features you used have only 64 abnormal ones and the rest are normal ones since you hardcoded it?

**Comment by tianyu0207** [comment]
> 1. Ok thanks, I'll provide another argument for an input file that has weak labels for the train set along with the already existing test label file argument.
> 2. By the way, does that mean the shanghai tech i3d features ([list/shanghai-i3d-train-10crop.list](list/shanghai-i3d-train-10crop.list)) you used have first 64 lines as abnormal ones and the rest are normal ones since you hardcoded it?

2. Yes. This is the same setup as the GCN-anomaly paper.
